src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -434,17 +434,35 @@ public virtual void WriteTo([NotNull] TextWriter writer, object value)
                     // an attribute value that may have been quoted with single quotes, must handle any double quotes
                     // in the value. Writing the value out surrounded by double quotes.
                     //
-                    // Do not combine following condition with check of escapeQuotes; htmlContent.ToString() can be
-                    // expensive when the IHtmlContent is created with a BufferedHtmlContent.
-                    var stringValue = htmlContent.ToString();
-                    if (stringValue.Contains("\""))
+                    // Be careful combining IHtmlContent with escapeQuotes= = true converting an IHtmlContent to
+                    // a string can be very expensive.
```

Comment:
Sure. I'll update this to reflect that it's not optimal. The code should document every instance of existential peril experienced by its' authors.